src/dbSta/src/dbReadVerilog.cc

```diff
@@ -363,29 +364,39 @@ void Verilog2db::makeDbModule(
       return;
     }
     if (hierarchy_) {
-      // make the module bterms
+      // make the module ports
       CellPortIterator* cp_iter = network_->portIterator(cell);
       while (cp_iter->hasNext()) {
         Port* port = cp_iter->next();
-        /* Ports are prefixed by instance name*/
         if (network_->isBus(port)) {
+          // make the bus port as part of the port set for the cell.
           const char* port_name = network_->name(port);
-          const char* cell_name = network_->name(cell);
-          int from = network_->fromIndex(port);
-          int to = network_->toIndex(port);
-          string key = "bus_msb_first ";
-          key += key + port_name + " " + cell_name;
-          key += port_name;
-          odb::dbBoolProperty::create(block_, key.c_str(), from > to);
+          dbModBTerm* bmodterm = dbModBTerm::create(module, port_name);
+          dbBusPort* dbbusport
+              = dbBusPort::create(module,
+                                  bmodterm,  // the root of the bus port
+                                  network_->fromIndex(port),
+                                  network_->toIndex(port));
+          bmodterm->setBusPort(dbbusport);
+          dbIoType io_type = staToDb(network_->direction(port));
+          bmodterm->setIoType(io_type);
+
           // Make a modbterm for each bus bit
-          int start_index = from < to ? from : to;
-          int end_index = from < to ? to : from;
-          for (int i = start_index; i <= end_index; i++) {
-            // use actual here
+          // Keep traversal in terms of bits
+          int from_index = network_->fromIndex(port);
+          int to_index = network_->toIndex(port);
+          bool updown = (from_index <= to_index) ? true : false;
+          int size
+              = updown ? to_index - from_index + 1 : from_index - to_index + 1;
+          for (int i = 0; i < size; i++) {
+            int ix = updown ? from_index + i : from_index - i;
             std::string bus_bit_port = port_name + std::string("[")
-                                       + std::to_string(i) + std::string("]");
-            dbModBTerm* bmodterm
+                                       + std::to_string(ix) + std::string("]");
+            dbModBTerm* modbterm
```

Comment:
Is there a reason to not have dbBusPort::create handle creating the scalar terms?

Reply:
The thinking here is not vector/scalar elements -- the dbBusPort does not own the modbterms nor allocate them. Think of the various bus structures (eg NetBus, Bundle, BusPort) as being simple "overlays" on the leaf level objects. The connectivity is always at the bit level. For example the bus port makes no assumption about the naming of the leaf objects.